staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -643,6 +643,8 @@ func ReasonForError(err error) metav1.StatusReason {
 	switch t := err.(type) {
 	case APIStatus:
 		return t.Status().Reason
+	case interface{ Unwrap() error }:
```

Comment:
The implementation of this method could use the [`error` package](https://golang.org/pkg/errors/), what do you think about that?:
```golang
func ReasonForError(err error) metav1.StatusReason {    
    if status := APIStatus(nil); errors.As(err, &status) {
        return status.Status().Reason
    }
    return metav1.StatusReasonUnknown
}
```

Reply:
Done